src/ray/core_worker/core_worker.cc

```diff
@@ -483,6 +483,13 @@ void CoreWorker::Shutdown() {
   is_shutdown_ = true;
   if (options_.worker_type == WorkerType::WORKER) {
     // Running in a main thread.
+    // Asyncio coroutines could still run after CoreWorker is removed because it is
```

Comment:
I guess we should terminate asyncio threads before shutting down other threads because asyncio threads can use deallocated instances (like gcs_client)?

Reply:
What @rkooo567 said is one possibility. Also we should terminate asyncio threads before boost fiber threads since the latter depends on the former.